### PR TITLE
cleaned up wrapper_generator.cpp a bit and added include guards to generated files

### PR DESCRIPTION
**Pull request by wmww**

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Bug #147 again

bors try

File: src/protocol/wrapper_generator.cpp [hunk 1]
```diff
@@ -27,26 +27,68 @@
...
     out << " * AUTOGENERATED - DO NOT EDIT" << std::endl;
     out << " *" << std::endl;
-    out << " * This header is generated by src/protocol/wrapper_generator.cpp" << std::endl;
+    out << " * This header is generated by src/protocol/wrapper_generator.cpp from " << input_file_path << std::endl;
     out << " * To regenerate, run the “refresh-wayland-wrapper” target." << std::endl;
     out << " */" << std::endl;
 }
 
-void emit_required_headers(std::string const& custom_header)
+// remove the path from a file path, laving only the base name
```
Comment (AlanGriffiths) [inline src/protocol/wrapper_generator.cpp]: "laving"?

File: src/protocol/wrapper_generator.cpp [hunk 2]
```diff
@@ -391,6 +451,8 @@ class Interface
             emit_indented_lines(out, "    ", {
                 { "uint32_t const max_version;" }
             });
+            if (!methods.empty())
+                out << std::endl;
```
Comment (AlanGriffiths) [inline src/protocol/wrapper_generator.cpp]: We don't need to force a flush(): "out << std::endl;" should be "out << '\n';"

File: src/protocol/wrapper_generator.cpp [hunk 3]
```diff
@@ -489,14 +537,16 @@ int main(int argc, char** argv)
     auto name_transform = [prefix](std::string protocol_name)
     {
         std::string transformed_name = protocol_name;
-        if (protocol_name.find(prefix) == 0)
+        if (protocol_name.find(prefix) == 0) // if the fist instance of prefix is at the start of protocol_name
```
Comment (gerboland) [inline src/protocol/wrapper_generator.cpp]: typo "fist"

File: src/protocol/wrapper_generator.cpp (hunk 2 above)
Comment (wmww) [reply src/protocol/wrapper_generator.cpp]: Is this case different from the other 60 instances of std::endl? Or should they all eventually be replaced by '\n'?

Comment (AlanGriffiths) [reply src/protocol/wrapper_generator.cpp]: Yes, just a case of not following a bad precedent

**Comment by RAOF** [comment]
Looks like a timeout failure?

bors r+

**Comment by wmww** [comment]
bors try

**Pull request merged**